pkg/parsers/filters/parse.go

```diff
@@ -65,6 +65,35 @@ func FromParam(p string) (Args, error) {
 	return args, nil
 }
 
+func (filters Args) MatchKVList(field string, sources map[string]string) bool {
+	fieldValues := filters[field]
+
+	//do not filter if there is no filter set or cannot determine filter
+	if len(fieldValues) == 0 {
+		return true
+	}
+
+	if sources == nil {
+		return false
+	}
+
+	for _, name2match := range fieldValues {
+		testKV := strings.SplitN(name2match, "=", 2)
+
+		for k, v := range sources {
+			if len(testKV) == 1 {
+				if k == testKV[0] {
+					return true
+				}
+			} else if k == testKV[0] && v == testKV[1] {
+				return true
```

Comment:
I disagree with the implementation of this method, in that it effectively treats multiple label filters as a `OR` rather than an `AND`. Maybe that's deliberate, but it contradicts how others filters are implemented.

Example:

```
$ docker run -d --label foo=bar busybox top
$ docker run -d --label foo=bar --label plop=gna busybox top

$ docker ps --filter status=exited --filter label=foo
# Empty output: it's an AND and no container is in status exited

$ docker ps --filter label=foo --filter label=plop
# Both containers are returned: it's an OR
```

I think the logic should be reversed, and that we should return `false` early when a given match **fails**, and ultimately return `true` at the end of the function.